src/main/java/org/folio/service/orders/flows/update/open/OpenCompositeOrderPieceService.java

```diff
@@ -95,31 +96,37 @@ private Future<List<Piece>> updatePieces(OpenOrderPieceHolder holder, RequestCon
 
   private Future<List<Piece>> createPieces(OpenOrderPieceHolder holder, CompositePurchaseOrder order, boolean isInstanceMatchingDisabled, RequestContext requestContext) {
     logger.debug("createPieces:: Trying to create pieces");
+
     List<Piece> piecesToCreate = new ArrayList<>(holder.getPiecesWithLocationToProcess());
     piecesToCreate.addAll(holder.getPiecesWithHoldingToProcess());
     piecesToCreate.addAll(holder.getPiecesWithoutLocationId());
 
-    var preparedPieces = piecesToCreate.stream()
+    List<Piece> preparedPieces = piecesToCreate.stream()
       .map(piece -> piece.withTitleId(holder.getTitleId()))
       .toList();
 
-    // Perform individual acq unit validations for each piece and open order inventory update operation before creating batch pieces
-    var validationFutures = preparedPieces.stream()
-      .map(piece -> openOrderUpdateInventory(piece, order, isInstanceMatchingDisabled, requestContext))
-      .toList();
+    // Collect pieces after validation
+    List<Future<Piece>> validationFutures = new ArrayList<>();
 
-    logger.info("createPieces:: Passed acq unit validation and open order '{}' inventory update", order.getId());
-    return collectResultsOnSuccess(validationFutures)
-      .compose(validatedPieces ->
-        pieceStorageService.insertPiecesBatch(validatedPieces, requestContext)
-          .map(PieceCollection::getPieces)
+    // Use chainCallCollect to sequentially process each piece with openOrderUpdateInventory.
+    return chainCall(preparedPieces, piece ->
+      openOrderUpdateInventory(piece, order, isInstanceMatchingDisabled, requestContext)
+        .onSuccess(validatedPiece -> validationFutures.add(Future.succeededFuture(validatedPiece)))
```

Comment:
better to use map here instead of onSuccess. Using map() instead of onSuccess() would ensure that the execution of the previous step is completed before proceeding